Engine/DataFeeds/Enumerators/BaseDataCollectionAggregatorEnumerator.cs

```diff
@@ -173,6 +173,16 @@ protected virtual void Add(T collection, BaseData current)
         {
             collection.Data.Add(current);
         }
+
+        /// <summary>
+        /// Adds all specified instances of <see cref="BaseData"/> to the current collection
+        /// </summary>
+        /// <param name="collection">The collection to be added to</param>
+        /// <param name="current">The data collection to be added</param>
+        protected virtual void AddAll(T collection, List<BaseData> current)
+        {
+            collection.Data = current;
```

Comment:
This method is called `AddAll` but it overwrites the existing collection. Perhaps this is more of a `SetData` method?